homeassistant/components/vallox/fan.py

```diff
@@ -111,7 +109,16 @@ def is_on(self) -> bool:
     def preset_mode(self) -> str | None:
         """Return the current preset mode."""
         vallox_profile = self.coordinator.data.profile
-        return VALLOX_PROFILE_TO_STR_REPORTABLE.get(vallox_profile)
+        return VALLOX_PROFILE_TO_PRESET_MODE_REPORTABLE.get(vallox_profile)
+
+    @property
+    def supported_features(self) -> FanEntityFeature:
+        """Flag supported features."""
+        features = FanEntityFeature.PRESET_MODE
+        if self.preset_mode in VALLOX_PRESET_MODES_SUPPORTING_FAN_SPEED:
+            features |= FanEntityFeature.SET_SPEED
```

Comment:
Supported features should not change during entity lifetime.

Reply:
Sad. I really liked how slider hides in lovelace for Fireplace preset...

Reply:
Removed